src/evaluation/searcher.h

```diff
@@ -316,20 +317,25 @@ class Searcher {
             Score score = 0;
             legalMoves++;
 
-            /*
-             * LMR
-             * https://wiki.sharewiz.net/doku.php?id=chess:programming:late_move_reduction
-             */
+            /* Late Move Reduction (LMR)
+             * https://wiki.sharewiz.net/doku.php?id=chess:programming:late_move_reduction */
             if (movesSearched == 0) {
                 score = -negamax(depth - 1, m_stackItr->board, -beta, -alpha);
             } else {
                 if (movesSearched >= spsa::fullDepthMove
-                    && depth >= spsa::lmrReductionLimit
-                    && !isChecked
                     && !move.isCapture()
                     && !move.isPromotionMove()) {
+
+                    const bool isGivingCheck = engine::isKingAttacked(m_stackItr->board);
+                    int8_t lmrReduction = spsa::lmrBaseReduction + getLmrReduction(depth, movesSearched);
+
+                    lmrReduction -= isChecked; /* reduce less when checked */
```

Comment:
Implicit conversion from bool to int here. What do you think about that?

Reply:
Made it explicit - you're right, it's better :+1: